hub/src/server/server.ts

```diff
@@ -213,6 +215,42 @@ export class HubServer {
       }
     }
 
+    const ifMatchTag = requestHeaders['if-match']
+    const ifNoneMatchTag = requestHeaders['if-none-match']
+
+    // only one match-tag header should be set
+    if (ifMatchTag && ifNoneMatchTag) {
+      throw new PreconditionFailedError('Request should not contain both if-match and if-none-match headers')
+    }
+
+    if (!this.driver.supportsETagMatching) {
+      // allow overwrites if tag is wildcard 
+      if (ifMatchTag && ifMatchTag !== '*') {
+        const currentETag = (await this.driver.performStat({
+          path: path,
+          storageTopLevel: address
+        })).etag
+
+        if (ifMatchTag !== currentETag) {
+          throw new PreconditionFailedError(
+            'The provided ETag does not match that of the resource on the server'
+          )
+        }
+      } else if (ifNoneMatchTag && ifNoneMatchTag === '*') {
+        // only proceed with writing file if the file does not already exist
+        const statResult = await this.driver.performStat({
+          path: path,
+          storageTopLevel: address
+        })
+
+        if (statResult.exists) {
+          throw new PreconditionFailedError('The entity you are trying to create already exists')
+        }
+      } else if (ifNoneMatchTag && ifNoneMatchTag !== '*') {
```

Comment:
I think this condition is supposed to be outside of the `if (!this.driver.supportsETagMatching)` condition scope. As in, next to the `if (ifMatchTag && ifNoneMatchTag)` check above.

Reply:
Why? Don't we only want to check for the existence of a file if the driver can't do it for us?

Reply:
I think without the check, this would result in some gaia hubs accepting `ifNoneMatchTag !== '*'`, while some hubs would not, depending on the driver implementation.